frontend/lib/ui/Table/Table.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { useNavigate } from "react-router-dom";
+
+import cls from "./Table.module.css";
+
+export interface TableProps {
+  id?: string;
+  children?: React.ReactNode;
+}
+
+export function Table({ id, children }: TableProps) {
+  return (
+    <table id={id} className={cls["table"]}>
+      {children}
+    </table>
+  );
+}
+
+Table.Header = ({ children }: { children: React.ReactNode[] }) => (
+  <thead>
+    <tr>{children}</tr>
+  </thead>
+);
+
+Table.Column = ({ children, number, width }: { children: React.ReactNode; number?: boolean; width?: string }) => (
+  <th className={number ? cls["number"] : undefined} style={width ? { width } : undefined}>
+    {children}
+  </th>
+);
+
+Table.Body = ({ children }: { children: React.ReactNode[] }) => <tbody>{children}</tbody>;
+
+Table.Row = ({ children }: { children: React.ReactNode[] }) => <tr>{children}</tr>;
+
+Table.LinkRow = ({ children, to }: { children: React.ReactNode[]; to: string }) => {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
Seems that ESLint does not recognise this `Table.LinkRow` as a React component

Reply:
Apparently this pattern of component declarations is not recommended, but there is a workaround for the lint error: https://github.com/facebook/react/issues/20700#issuecomment-1919251632

Reply:
Do you mean the <Table.Cell> part?
I've been slowly extracting the separate components for some (like ProgressList)
The reason to choose these type of exports is to have some namespacing and reduce the amount of imports required, with the goal of improving readability for "new" coders.

I have no feelings towards removing this "code style" but curious about the reasons.

Reply:
I've updated it to have the function definition separated, to make it more clear (also to the linter) that they are React components, thanks for the suggestion!

And I prefer this dot notation as well for discoverability in the IDE (autocomplete).